client-runtime/src/main/java/com/microsoft/rest/v2/SwaggerMethodParser.java

```diff
@@ -220,7 +220,7 @@ public String httpMethod() {
     public String scheme(Object[] swaggerMethodArguments) {
         final String substitutedHost = applySubstitutions(rawHost, hostSubstitutions, swaggerMethodArguments, UrlEscapers.urlPathSegmentEscaper());
         final String[] substitutedHostParts = substitutedHost.split("://");
-        return substitutedHostParts.length < 1 ? "https" : substitutedHostParts[0];
+        return substitutedHostParts.length < 1 ? null : substitutedHostParts[0];
```

Comment:
Is the plan now that we're not going to pick a default scheme for users? As in, if Swagger didn't provide it and the host parameter value didn't provide it, then we just pass it on to the HTTP client and it does whatever it decides to do?

Reply:
Yes. We have some tests in autorest.java that will run into this very scenario, which is why I created the ProtocolPolicy RequestPolicy. If no protocol is provided and the HttpRequest hits the HttpClient, then the HttpClient will most likely throw an exception.